File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/non-vul/code_before41.c
// From cwe-snippets, snippets_90/non-compliant/C/cwe-0252/char_fgets_01.c

static void func()
{
    {
        char dataBuffer[100] = "";
        char * data = dataBuffer;
        printLine("Please enter a string: ");
        if (fgets(data, 100, stdin) == NULL)
        {
            printLine("fgets failed!");
            exit(1);
        }
        printLine(data);
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before41.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0329/basic_01.java

private void func() throws Throwable
{

    byte[] text = "asdf".getBytes("UTF-8");

    KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
    keyGenerator.init(128);
    SecretKey key = keyGenerator.generateKey();

    Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");

    int blockSize = cipher.getBlockSize();
    byte[] initializationVector = new byte[blockSize];
    SecureRandom secureRandom = new SecureRandom();
    secureRandom.nextBytes(initializationVector);
    IvParameterSpec ivParameterSpec = new IvParameterSpec(initializationVector);

    cipher.init(Cipher.ENCRYPT_MODE, key, ivParameterSpec);

    IO.writeLine(IO.toHex(cipher.doFinal(text)));

}
